ui/src/main/java/edu/wpi/grip/ui/pipeline/PipelineController.java

```diff
@@ -278,13 +273,13 @@ public void onStepMoved(StepMovedEvent event) {
     @Subscribe
     public void onConnectionAdded(ConnectionAddedEvent event) {
         // Add the new connection view
-        platform.runAsSoonAsPossible(() -> addConnectionView(event.getConnection()));
+        addConnectionView(event.getConnection());
     }
 
     @Subscribe
     public void onConnectionRemoved(ConnectionRemovedEvent event) {
         // Remove the control that corresponds with the connection that was removed
-        platform.runAsSoonAsPossible(() -> {
+        PlatformImpl.runAndWait(() -> {
```

Comment:
We *want* it to wait on the JavaFX thread, otherwise there's flickering.

> Basically, I don't trust runAndWait at all. 

I think that's a bit unfair. `runAndWait` would only deadlock if we post an event inside of it. It's totally appropriate to use for cases where we're only manipulating JavaFX components (not grip objects) from an arbitrary thread and we want it to complete atomically.

There's nothing untrustworthy about it, it just doesn't interact well with the unusual behavior of `eventBus.post()`.

Trust me, I've tested this code in every possible situation, and I've spent some time reasoning about how this needs to work. Doing it this way removes some annoying flickering, and I think that will have a positive impact on users trying this for the first time.